openssh/map.jinja

```diff
@@ -3,205 +3,59 @@
-            ~ ", merge: strategy='"
-            ~ _strategy
-            ~ "', lists='"
-            ~ _config["merge_lists"]
-            ~ "'"
-          ) %}
-
-{#-       Keep values under each root key when there are more than one #}
-{%-       if config_get_roots|length > 1 %}
-{%-         set _config_get = { _config_root: _config_get } %}
-{%-       endif %}
-{%-       do _config.update(
-            {
-              "stack": salt["slsutil.merge"](
-                _config["stack"],
-                _config_get,
-                strategy=_strategy,
-                merge_lists=_config["merge_lists"],
-              )
-            }
-          ) %}
-{%-     endfor %}
-{%-   else %}
-{#-     Lookup the grain/pillar/... #}
-{#-     Fallback to use the source name as a direct filename #}
-{%-     set map_values = salt["config.get"](map_source, []) %}
-
-{#-     Mangle `map_source` to use it as literal path #}
-{%-     if map_values | length == 0 %}
-{%-       set map_source_parts = map_source.split("/") %}
-{%-       set map_source = map_source_parts[0:-1] | join("/") %}
-{%-       set map_values = map_source_parts[-1].rstrip(".yaml") %}
-{%-     endif %}
-
-{#-     Some configuration return list #}
-{%-     if map_values is string %}
-{%-       set map_values = [map_values] %}
-{%-     endif %}
-
-{%-     for map_value in map_values %}
-{%-       set yamlfile = [
-            map_sources_dir,
-            map_source,
-            map_value ~ ".yaml",
-          ]
-          | join("/")
-          %}
-{%-       do salt["log.debug"]("map.jinja: load parameters from file " ~ yamlfile) %}
-{%-       load_yaml as loaded_values %}
-{%-         include yamlfile ignore missing %}
-{%-       endload %}
-
-{%-       if loaded_values %}
-{#-         Merge loaded values on the stack #}
-{%-         do salt["log.debug"]("map.jinja: merge parameters from " ~ yamlfile) %}
-{%-         do _config.update(
-              {
-                "stack": salt["slsutil.merge"](
-                  _config["stack"],
-                  loaded_values.get("values", {}),
-                  strategy=loaded_values.get("strategy", "smart"),
-                  merge_lists=loaded_values.get("merge_lists", False)
-                  | to_bool,
-                )
-              }
-            ) %}
-{%-       endif %}
-{%-     endfor %}
-{%-   endif %}
-{%- endfor %}
+      "map.jinja: load parameters from sources"
+      ~ ":\n" ~ map_sources
+      | yaml(False)
+    ) %}
+
+{#- Load formula parameters values #}
+{%- set _formula_matchers = ["defaults.yaml"] + map_sources %}
+{%- set _formula_settings = mapstack(
+      matchers=_formula_matchers,
+      dirs=[formula_param_dir],
+      defaults={"values": {},
+                "merge_strategy": salt["config.get"](tplroot ~ ":strategy", None),
+                "merge_lists": salt["config.get"](tplroot ~ ":merge_lists", False),
+               },
+      log_prefix="map.jinja: ",
+    )
+    | load_yaml
+    %}
```

Comment:
Minor style consistency issue, we're closing Jinja tags on the last line for the rest of the lines, so let's maintain that here:

```suggestion
    | load_yaml %}
```

If it needs to be maintained on a separate line, I'd propose moving it to the same indentation as the starting tag  instead:

```suggestion
    | load_yaml
%}
```